app/controllers/users_controller.rb

```diff
@@ -4,11 +4,14 @@ class UsersController < ApplicationController
 
   def new
     @user = User.new
+    @invitation = Invitation.find_by token: params[:token]
   end
 
   def create
-    @user = User.create user_params
-    if @user.valid?
+    @invitation = Invitation.find_by token: user_params[:invite_token]
```

Comment:
This step will be executed on every registration. It would be helpful to have it delayed and see if the user record is first valid, then you should check if the invite_token param is present to have some kind of `hand_invitation` workflow allowed after user is saved successfully. `find_by` hits the db so its good to avoid that if you don't have to.